### PR TITLE
Reduce noise of webhook handling for many jobs

### PR DESCRIPTION
**Pull request by DavidTanner**
Webhook logs should only be generated once, and each job doesn't need to handle the processing of the body to an object.

### CODE REVIEW

**Comment by DavidTanner** [comment]
@arcivanov would you give this a look and see if I am missing something?  I believe an anonymous connection to GitHub can parse the body just fine.  Then the work isn't duplicated so much.

File: src/main/java/org/jenkinsci/plugins/ghprb/GhprbRootAction.java [hunk 1]
```diff
@@ -84,15 +87,52 @@ public void doIndex(StaplerRequest req, StaplerResponse resp) {
...
-            try {
-                webHook.handleWebHook(event, payload, body, signature);
-            } catch (Exception e) {
-                logger.log(Level.SEVERE, "Unable to process web hook for: " + webHook.getProjectName(), e);
+        logger.log(Level.INFO, "Got payload event: {0}", event);
+        
+        try {
+            GitHub gh = GitHub.connectAnonymously();
+            
+            if ("issue_comment".equals(event)) {
```
Comment (arcivanov) [inline src/main/java/org/jenkinsci/plugins/ghprb/GhprbRootAction.java]: Finally! :+1:

File: src/main/java/org/jenkinsci/plugins/ghprb/GhprbRootAction.java [hunk 2]
```diff
@@ -84,15 +87,52 @@ public void doIndex(StaplerRequest req, StaplerResponse resp) {
...
+                        webHook.handlePR(pr, body, signature);
+                    } catch (Exception e) {
+                        logger.log(Level.SEVERE, "Unable to process web hook for: " + webHook.getProjectName(), e);
+                    }
+                }
+            } else {
+                logger.log(Level.WARNING, "Request not known");
             }
+
+        } catch (IOException e1) {
```
Comment (arcivanov) [inline src/main/java/org/jenkinsci/plugins/ghprb/GhprbRootAction.java]: Does Jenkins logger really fail with IOException?

File: src/main/java/org/jenkinsci/plugins/ghprb/GhprbRootAction.java [hunk 3]
```diff
@@ -84,15 +87,52 @@ public void doIndex(StaplerRequest req, StaplerResponse resp) {
...
             return;
         }
+        
 
-        for (GhprbWebHook webHook : getWebHooks()) {
-            try {
-                webHook.handleWebHook(event, payload, body, signature);
-            } catch (Exception e) {
-                logger.log(Level.SEVERE, "Unable to process web hook for: " + webHook.getProjectName(), e);
+        logger.log(Level.INFO, "Got payload event: {0}", event);
```
Comment (arcivanov) [inline src/main/java/org/jenkinsci/plugins/ghprb/GhprbRootAction.java]: Maybe move this to DEBUG?

File: src/main/java/org/jenkinsci/plugins/ghprb/GhprbRootAction.java (hunk 2 above)
Comment (DavidTanner) [inline src/main/java/org/jenkinsci/plugins/ghprb/GhprbRootAction.java]: The exception comes from GitHub.connectAnonymously()

**Comment by arcivanov** [comment]
LGTM :+1:

**Pull request merged**